roofit/multiprocess/src/PriorityQueue.cxx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Project: RooFit
+ * Authors:
+ *   PB, Patrick Bos, Netherlands eScience Center, p.bos@esciencecenter.nl
+ *   IP, Inti Pelupessy, Netherlands eScience Center, i.pelupessy@esciencecenter.nl
+ *
+ * Copyright (c) 2022, CERN
+ *
+ * Redistribution and use in source and binary forms,
+ * with or without modification, are permitted according to the terms
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)
+ */
+
+#include "RooFit/MultiProcess/JobManager.h"
+#include "RooFit/MultiProcess/Messenger.h"
+#include "RooFit/MultiProcess/PriorityQueue.h"
+#include "RooFit/MultiProcess/types.h"
+
+#include <algorithm> // std::copy
+
+namespace RooFit {
+namespace MultiProcess {
+
+/** \class PriorityQueue
+ * \brief Queue that orders tasks according to specified task priorities.
+ */
+
+/// See Queue::pop.
+bool PriorityQueue::pop(JobTask &job_task)
+{
+   if (queue_.empty()) {
+      return false;
+   } else {
+      job_task = queue_.top().jobTask;
+      queue_.pop();
+      return true;
+   }
+}
+
+/// See Queue::add.
+void PriorityQueue::add(JobTask job_task)
+{
+   if (JobManager::instance()->process_manager().is_master()) {
+      JobManager::instance()->messenger().send_from_master_to_queue(M2Q::enqueue, job_task.job_id, job_task.state_id,
+                                                                    job_task.task_id);
+   } else if (JobManager::instance()->process_manager().is_queue()) {
```

Comment:
I made the bad experience in other parts of RooFit that functions that return static variables are quite expensive.
Maybe do this at the beginning of the function to only call the static function once?
```
auto& jobManager = *JobManager::instance();
```
I don't know if it makes a performance difference here, but just as a matter of precaution and because it also makes the code a bit more readable.

Reply:
Makes sense! Indeed, probably not a bottleneck, but good practice.